api/src/main/java/ai/djl/inference/Predictor.java

```diff
@@ -144,6 +144,9 @@ public O predict(I input) throws TranslateException {
     protected NDList predictInternal(TranslatorContext ctx, NDList ndList)
             throws TranslateException {
         logger.trace("Predictor input data: {}", ndList);
+        if (ndList.isEmpty()) {
+            return new NDList();
+        }
```

Comment:
@frankfliu @bryanktliu why is this needed in Predictor? Just FYI that this is causing failures in DJL Serving https://github.com/deepjavalibrary/djl-serving/actions/runs/12792959567/job/35664630152?pr=2663